### ISSUE TITLE
Required @OA\Info() not found for v4.

### ISSUE THREAD

**Issue by ShimamuraSeiya** [zircote/swagger-php#1005]
This is error message.
```
WarningUninitialized string offset 1
WarningUninitialized string offset 1
WarningUninitialized string offset 1
WarningUninitialized string offset 1
WarningUninitialized string offset 1
WarningUninitialized string offset 1
Warning: Required @OA\Info() not found
```

**Comment by DerManoMann** [comment]
Any chance of some code to reproduce?
Also, have you checked that all annotations are associated with classes, etc? Using reflection means you might need an empty class/interface for your 'global' annotations...
Something like
```
/**
 * @OA\Info(...)
 */
class SomeApi {}
```

**Comment by DerManoMann** [comment]
Should be fixed in 4.0.2 for sure

**Comment by williamdes** [comment]
For other users searching, I had `Required @OA\Info() not found` and the only thing that worked is re-writing the analysis part.
Using: https://github.com/zircote/swagger-php/blob/master/docs/Migrating-to-v4.md#improvements-to-the-generator-class

And some tweaks to improve the provided example

```diff
         $generator = new Generator();
         $analyser = $generator->getAnalyser();
-        $context = new Context();
-        $analysis = new Analysis([], $context);
+        $analyser->setGenerator($generator);
+
         $exclude = null;
 
         // Crawl directory and find all files
         $finder = Util::finder(app_path(), $exclude);
-        foreach ($finder as $file) {
-            $analysis->addAnalysis($analyser->fromFile($file->getPathname(), $context));
-        }
-
-        // Post processing
-        $analysis->process($generator->getProcessors());
-        // Validation (Generate notices & warnings)
+
+        $analysis = $generator->withContext(
+            function (Generator $generator, Analysis $analysis, Context $context) use ($analyser, $finder): Analysis {
+
+                foreach ($finder->getIterator() as $file) {
+                    $analysis->addAnalysis($analyser->fromFile($file->getPathname(), $context));
+                }
+
+                $analysis->process($generator->getProcessors());
+                return $analysis;
+            }
+        );
+
        $analysis->validate();
```
